src/staking/unbondingAccount.ts

```diff
@@ -123,26 +133,36 @@ export class UnbondingAccount {
   constructor(
     public program: Program<StakeIdl>,
     public address: PublicKey,
-    public unbondingAccount: UnbondingAccountInfo
+    public unbondingAccount: UnbondingAccountInfo,
+    public stakePool: StakePool
   ) {
     this.tokens = new BN(0)
+    if (stakePool.stakePool.unbonding.tokens !== new BN(0) && stakePool.stakePool.unbonding.shares !== new BN(0)) {
+      this.tokens = unbondingAccount.shares
+        .mul(stakePool.stakePool.unbonding.tokens)
+        .div(stakePool.stakePool.unbonding.shares)
+    }
```

Comment:
I just got major deja vu by looking at this code 😆 

Gotta use the BN functions to do math unfortunately, `!==` `==` `<` `>`  don't work. And you only need to check for one of these being zero if you only care about tokens

Also you can avoid allocating a BN object

```suggestion
    if (stakePool.stakePool.unbonding.shares.gt(new BN(0))) {
      this.tokens = unbondingAccount.shares
        .mul(stakePool.stakePool.unbonding.tokens)
        .div(stakePool.stakePool.unbonding.shares)
    } else {
      this.tokens = new BN(0);
    }
```

As an aside after looking at sharesToTokensUnbonded it doesn't check for div by 0 because it's impossible in the program. I think it is possible if you load the accounts in the wrong order, so we need it